src/core/inlines.js

```diff
@@ -100,25 +106,21 @@ export function run(conf) {
             );
           } else {
             const { type, illegal } = refTypeFromContext(ref, txt.parentNode);
-            ref = ref.replace(/^(!|\?)/, "");
-            df.appendChild(document.createTextNode("["));
-            const refHref = `#bib-${ref.toLowerCase()}`;
-            const cite = hyperHTML`<cite><a class="bibref" href="${refHref}">${ref}</a></cite>`;
-            df.appendChild(cite);
-            df.appendChild(document.createTextNode("]"));
-
-            if (illegal && !conf.normativeReferences.has(ref)) {
+            const cite = renderInlineCitation(ref);
+            const cleanRef = ref.replace(/^(!|\?)/, "");
```

Comment:
Previously the code removed `!`/`?` first and this now doesn't. Would there be a behavior change?

Reply:
Well spotted. It's now done in by `renderInlineCitation()` - so same behavior.